### ISSUE TITLE
Scratch "set PWM" command is not executed consistently

### ISSUE THREAD

**Issue by gmenegoz** [yokobond/pxt-scratch-microbit-more#1]
Hi,
we (me and @virivigio) used your pxt microbit firmware together with the MicrobitMore Scratch extension finding that the _Set PWM_ command sent from the Scratch extension is not executed consistently. The Scratch extension sends the command and the command is always received but a LED connected on a PWM pin doesn't show consistent behaviour (the same LED controlled by _Set digital_ block works correctly)

**Steps to reproduce the behavior:**
1. Load firmware on microbit
2. Load microbit-more Scratch extension on a local instance of the Scratch gui
3. Connect two LEDs to pins 1 and 2 (and ground of course)
3. Turn on Scratch Link and connect Microbit
4. Use _Set PWM_ blocks as in the attached sb3 (to control LEDs attached on pins 1 and 2 using keyboard)
5. Press left and right arrow keys
6. Notice that the LED controlled via _Set digital_ block works consistently whereas the LED controlled via the _Set PWM_ behaves randomly: only after many keypresses the command is executed (thus proving that the LED is working and wiring is correct)

**Hardware connected to micro:bit**
- USB-powered microbit connected via BLE (Scratch-Link)
- LEDs connected to pins 1 and 2

In the attached sb3 you can find a Scratch code that asks both LEDs  to turn on and off when right and left keys are pressed; the LED matrix is also used to check that the message is received. The two LEDs should be always both on or off

**Screenshots**
I attach some screenshots of our setup: sometimes only the "set digital" controlled LED is turned on (or viceversa)
![photo_2020-05-03_18-07-43](https://user-images.githubusercontent.com/10050155/80919223-23354e80-8d69-11ea-9ca7-af0c01cd2558.jpg)
![photo_2020-05-03_18-07-45](https://user-images.githubusercontent.com/10050155/80919225-24ff1200-8d69-11ea-924a-0cd7b4160d15.jpg)
![photo_2020-05-03_18-07-47](https://user-images.githubusercontent.com/10050155/80919227-26303f00-8d69-11ea-80d7-25ddb77e9d02.jpg)
![photo_2020-05-03_18-07-50](https://user-images.githubusercontent.com/10050155/80919230-27616c00-8d69-11ea-8253-1809ed2d7f41.jpg)
![photo_2020-05-03_18-07-54](https://user-images.githubusercontent.com/10050155/80919231-29c3c600-8d69-11ea-80eb-217f78c17f86.jpg)

**Scratch:**
 - OS: both Win10 and MacOsX
 - Browser: Chrome and Firefox on Win10, Chrome and Safari on MacOs X

**Additional context**
Sb3 project to reproduce the issue:

[BlinkScratchDigital.zip](https://github.com/yokobond/pxt-scratch-microbit-more/files/4570924/BlinkScratchDigital.zip)

**Comment by virivigio** [comment]
Hi there, just a quick update. What Giuseppe described happens with version 0.3.2; then we discovered version 0.4.0 and gave it a try: actually the inconsistent behaviour still applies but the light is much more responsive than before. On an average, the pwm level will follow the ble command nine times out of ten.
Does this ring you a bell on where the problem might be? We'll be happy to help if you can suggest a strategy.

**Comment by yokobond** [comment]
Scratch-microbit-more firmware and scratch-extension are updated to v0.4 recently and has been changed large amount of code. 
Then I run your scratch project on the v0.4 and the voltage of pin-2 raised up to 3.2V every up-arrow key. 
Please use v0.4 online [scratch-microbit-more](https://yokobond.github.io/scratch-microbit-more/). 
(I'm working on build v0.4 app for Win10 and Mac.)

**Comment by virivigio** [comment]
Hi @yokobond  thank you for trying yourself our code.
We tried version 4:
- makecode with "pxt-scratch-microbit-more" loaded via github url
and
- local makecode with "pxt-scratch-microbit-more" loaded via pxt.json and local filesystem
- scratch online at your site

workflow:
up key -> ok
down key -> ok
up key -> ok
down key -> ok
up key -> ok
down key -> KO
up key -> ok
down key -> ok
up key -> ok
down key -> ok
up key -> ok
down key -> ok
up key -> ok
down key -> ok
up key -> ok
down key -> ok
up key -> KO
down key -> ok
and so on...

this time we also checked the voltage.
what are we missing?

Thank you for your work and support
